app/mender.go

```diff
@@ -345,24 +348,42 @@ func (m *Mender) CheckUpdate() (*datastore.UpdateInfo, menderError) {
 	return ur.UpdateInfo, nil
 }
 
-func (m *Mender) handleControlMap(data *client.UpdateResponse) error {
-	if data.UpdateControlMap != nil {
-		if data.UpdateControlMap.ID != "" {
-			if data.UpdateControlMap.ID != data.UpdateInfo.ID {
+// RefreshServerUpdateControlMap updates the control maps from the server during
+// a deployment.
+func (m *Mender) RefreshServerUpdateControlMap(deploymentID string) error {
+	cm, err := client.GetUpdateControlMap(
+		m.api,
+		m.Config.Servers[0].ServerURL,
+		deploymentID,
+	)
+	if err != nil {
+		return err
+	}
+
+	return m.HandleControlMap(deploymentID, cm)
+}
+
+func (m *Mender) HandleControlMap(
+	deploymentID string,
+	updateControlMap *updatecontrolmap.UpdateControlMap,
+) error {
+	if updateControlMap != nil {
+		if updateControlMap.ID != "" {
+			if updateControlMap.ID != deploymentID {
 				return NewTransientError(
-					fmt.Errorf("Mismatched control map ID: %s and deployment ID: %s",
-						data.UpdateControlMap.ID, data.UpdateInfo.ID))
+					fmt.Errorf(errControlMapIDMismatchF,
```

Comment:
Is it necessary to switch to a constant? It's only used here, and it's harder to read than the original.

Reply:
Actually, our new vet checks made the line too long, that's why ...

Reply:
Aha!